packages/server/postgres/getPgClient.ts

```diff
@@ -0,0 +1,10 @@
+import getPgPool from './getPgPool'
```

Comment:
i see the two as distinct. `getPgPool` gives access to the pool. you can then call `pool.query` which will get any available client from the pool then automatically release it after you call a single query.

`pool.connect` returns a single client which you can then use to do multiple queries. useful for transactions or when you need to ensure multiple queries go on the same connection in general. if using this pattern with vanilla `pg`, you have to remember to release the client manually.

`getPgPool` would be useful for usage of the first type, and `getPgClient` would be useful for usage of the second type. 

in `pgTyped`, the API requires passing a client into the generated function, so that's why `getPgClient` was used in this instance. i kept the two getters as separate as we're setting up the postgres API in general, but it could be that we only end up using the client getter in practice.

Reply:
that'd be a dealbreaker for pgtyped if true!
can you show me where you're seeing it requires a client & not a pool? from what i see, the param def looks like:
```ts
export interface IDatabaseConnection {
    query: (query: string, bindings: any[]) => Promise<{
        rows: any[];
    }>;
}
```

lets get rid of `getPgClient` all together, it really only does 1 thing & that 1 thing is dangerous (provide an unterminating connection). when the time for transactions comes, those transactions can grab a connection in their own functions.

Reply:
yeah you're right. you can pass in either a pool or client. i was just going by the docs instead of looking at the types and source code.

Reply:
> lets get rid of getPgClient all together, it really only does 1 thing & that 1 thing is dangerous (provide an unterminating connection). when the time for transactions comes, those transactions can grab a connection in their own functions.

i was seeing the client getter as more placeholder for work in progress. that's a good point though, somebody looking at the code as-is could be encouraged to do bad behavior.

Reply:
client getter deleted